### DISCUSSION TITLE
qs module depth limit issue

### DISCUSSION THREAD

**Discussion by salman3001** [adonisjs/core#4279] [Help] [answered]
hi there,

Please help me with this. i have some graphql type api with adonis. Means i can fetch or filter nested relationships by sending the query string.  somthing like this...  

```
const { data, loading, onRequest, pagination, tableRef } = useGetTableData(
  'address/cities',
  {
    populate: {
      state: {
        fields: ['name', 'id', 'country_id'],
        populate: {
          country: {
            fields: ['name', 'id', 'continent_id'],
            populate: {
              continent: {
                fields: ['name', 'id'],
              },
            },
          },
        }
      }
    },
  },
);
```

this is using axios under the hood. so the final query string in the request is like this.

```
sortBy: id
descending: false
page: 1
rowsPerPage: 10
search[name]: 
relationFilter[state][field]: id
relationFilter[state][filter][country][field]: id
relationFilter[state][filter][country][filter][continent][field]: id
relationFilter[state][filter][country][filter][continent][value]: 1
populate[state][fields][0]: name
populate[state][fields][1]: id
populate[state][fields][2]: country_id
populate[state][populate][country][fields][0]: name
populate[state][populate][country][fields][1]: id
populate[state][populate][country][fields][2]: continent_id
populate[state][populate][country][populate][continent][fields][0]: name
populate[state][populate][country][populate][continent][fields][1]: id
```
 it is working fine till 5 levels deep. but beyond that the query string is not parsed properly. so in this example the populate 'continent' is parsed like this

```
{
  field: 'id',
  filter: { continent: { '[field]': 'id', '[value]': '1' } }
}
```

you can see object key is string with brackets ` { '[field]': 'id', '[value]': '1' }`.

i know it is related to qs module parse depth limit config. So i tried to set the config for body parser  and increased the depth to 10. But that doesnt help. If any one can guide me? it will be lot of help for me.

**Comment by abdalem** [comment]
Hi everyone. I have the similar issue !
![image](https://github.com/adonisjs/core/assets/19971392/a9266b4d-fc27-4680-a29b-0d56841259e9)
![image](https://github.com/adonisjs/core/assets/19971392/548a664a-fa21-4923-9033-85c178d908bb)

**Comment by salman3001** [comment, marked as answer]
Solution  --- https://github.com/adonisjs/core/discussions/4293#discussioncomment-7667037
